U/lynnej/replace factorial

rhiannonlynne opened the pull request:
Replace numpy.math.factorial with math.factorial in function to smooth borosilicate glass values. 

Also update to setuptools_scm > 8 and consolidate python packaging files while we're at it.

rhiannonlynne committed:
Replace numpy.math.factorial with math.factorial

rhiannonlynne committed:
Update project packaging

rhiannonlynne committed:
Black and isort

ehneilsen reviewed (commented):
Looks fine, and I think the switch from `np.where()[0].size>0` to `np.any` in asorted places makes things clearer, but I don't understand why it was necessary. Can `deriv` be an array now?

ehneilsen reviewed (approved):
I meant to approve it before, but accidentally commented instead.

rhiannonlynne commented:
`deriv` actually isn't ever changed in this code, at least how it's used .. in theory it could be,  but I think it's fine for it to always be an integer and not an array. 
It's actually been a while since I modified the np.any parts (this was a change I must have made on my local copy a long while ago). I think the problem was that I am testing if there are any values <0 (which can indicate slightly bad measurements and should just be interpolated), and then out of those, testing if any are < belowzerothreshold (which is a more negative number and indicates properly bad data) -- and in the way I was testing that, it was written sometimes inconsistently and sometimes this failed. So I just replaced them all with the simpler to read syntax.
